package.json

```diff
@@ -105,13 +105,17 @@
 		"rules": {
 			"no-alert": "off",
 			"import/no-unassigned-import": "off",
-			"import/no-unresolved": "warn"
+			"import/no-unresolved": "warn",
+			"import/named": "warn"
 		},
 		"overrides": [
 			{
 				"files": "**/*.+(ts|tsx)",
 				"extends": "xo-typescript",
 				"rules": {
+					"@typescript-eslint/no-misused-promises": "off",
```

Comment:
Annoying: It doesn't let us use `async` functions as event handlers.

Reply:
While it's convenient, my reasoning is that using async there makes it hard to see that the event handler function will cause an uncaught rejection if the `await` rejects. For example, using async in a `new Promise` can cause hard to track down issues.

And when people see:

```js
[1, 2, 3].forEach(async value => {
  await doSomething(value);
});
```

they might be fooled into thinking that it awaits for each iteration, which it does not.

I'm happy to discuss this though. Maybe there some additional config that could be added to the rule?

Reply:
I understand the value in that case, but I don't have a solution for our use-case: our handlers are `async` just because `Promise` handling is great, not because anyone expects `addEventListener` to `await` them.

If we enable this rule, we have to write listeners as:

```js
function handler() {
	(async () => {
		document.body.append(await fetchDom(...));
	})();
}
```

Reply:
A more useful related rule would be the opposite: always await async functions. Does this exist already?


```js
async function set(){...}
async function get(){...}
set(1); // Forgot `await`
await get();
```

Reply:
> A more useful related rule would be the opposite: always await async functions. Does this exist already?

This? https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md

Reply:
Almost, `returnsPromise()` is not awaited even though it returns a promise. It seems that it requires `await` only on variables